target/linux/ath79/nand/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+[ -e /lib/firmware/$FIRMWARE ] && exit 0
+
+. /lib/functions/caldata.sh
+
+board=$(board_name)
+
+case "$FIRMWARE" in
+"ath10k/cal-pci-0000:00:00.0.bin")
+	case $board in
+	glinet,gl-ar750s-nor|\
+	glinet,gl-ar750s-nor-nand)
+		caldata_extract "art" 0x5000 0x844
+		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary art 0x0) +1)
```

Comment:
Have you checked whether there is a valid MAC address in 0x5006? Since they seem to provide the address in the correct location for wmac (0x1002), I'd suspect they also do for ath10k.
Just quickly do
```
. /lib/functions/system.sh
. /lib/functions.sh
mtd_get_mac_binary art 0x5006
```
If the correct address is there, you can remove the entire ath10k_patch_mac line. If you test that anyway, you can also remove the <&art 0x1002> in DTS, as wmac address should be read automatically, too.

Reply:
Reconfirmed that it is not present there
```
root@OpenWrt:~# . /lib/functions/system.sh
root@OpenWrt:~# . /lib/functions.sh
root@OpenWrt:~# mtd_get_mac_binary art 0x5006
ff:ff:ff:ff:ff:ff
```
In reference to the question about 0x6 (and other locations within ART), communication with GL.iNet technical staff indicated that only the 0x0 value was robust.

https://forum.gl-inet.com/t/ar-300m-ar-300m-lite-art-check-requested/5506

Reply:
In light of _[OpenWrt-Devel] ath79: remove redundant mtd-mac-address for wmac_ https://patchwork.ozlabs.org/patch/1184056/

preemptively removed `mtd-mac-address = <&art 0x1002>;`

and confirmed Tested-by: on that patch

Reply:
Thanks for testing. So, ath10k_patch_mac is necessary.

Concerning 0x6, their answer is quite vague, but I'm fine with your current solution there, too.

Reply:
Thanks for poking me. I'll re-check, as I think that I had a GL-AR300M plugged in, which is an ath9k device only.

Reply:
_(agreed resolved as ART does not contain the MAC addresses sought)_